pkg/collector/corechecks/gpu/stats.go

```diff
@@ -0,0 +1,236 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build linux
+
+package gpu
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/gpu/model"
+	"github.com/DataDog/datadog-agent/pkg/metrics/event"
+	sectime "github.com/DataDog/datadog-agent/pkg/security/resolvers/time"
+)
+
+const (
+	metricNameMemory = "gpu.memory"
+	metricNameUtil   = "gpu.utilization"
+	metricNameMaxMem = "gpu.max_memory"
+)
+
+// StatsProcessor is responsible for processing the data from the GPU eBPF probe and generating metrics from it
+type StatsProcessor struct {
+	// key is the key of the stream this processor is processing
+	key *model.StreamKey
+
+	// totalThreadSecondsUsed is the total amount of thread-seconds used by the GPU in the current interval
+	totalThreadSecondsUsed float64
+
+	// sender is the sender to use to send metrics
+	sender sender.Sender
+
+	// gpuMaxThreads is the maximum number of threads the GPU can run in parallel, for utilization computations
+	gpuMaxThreads int
+
+	// lastCheck is the last time the processor was checked
+	lastCheck time.Time
+
+	// measuredInterval is the interval between the last two checks
+	measuredInterval time.Duration
+
+	// timeResolver is the time resolver to use to resolve timestamps
+	timeResolver *sectime.Resolver
```

Comment:
can we extract this time pkg from security to some shared location? (in a separate PR of course)
cc @safchain

Reply:
Sounds good to me.

Reply:
https://datadoghq.atlassian.net/browse/EBPF-579